swarm/api/manifest.go

```diff
@@ -62,6 +62,11 @@ func readManifest(manifestReader storage.LazySectionReader, hash storage.Key, dp
 
 	// TODO check size for oversized manifests
 	size, err := manifestReader.Size(quitC)
+	if err != nil { // size == 0
+		// can't determine size means we don't have the root chunk
+		glog.V(logger.Detail).Infof("Manifest %v not found.", hash.Log())
+		return
```

Comment:
To clarify, the error should probably be `ErrManifestNotFound` which you can define as a top-level variable.